### PR TITLE
Reduce visual footprint of DAG Alerts in Slack

### LINKED ISSUE #3987: Reduce/Improve visual footprint of DAG Alerts in Slack
## Problem

Our Slack DAG alerts are phenomenal! However, when there are many of them their overall size and length can make it difficult to scan through them efficiently. Here is an example of one:

image

I would like to propose some simplifications and improvements. 

## Description

image

- Make the words "View logs" a link rather than printing the link itself. Move to the bottom
- Switch the Exception type and Exception line order. Since the type is typically a one-liner and the exeption itself will now be the only multiline field, it is nice to have at the bottom.

### PR DESCRIPTION
**Pull request by nicoepp**
## Fixes

Fixes #3987 by @zackkrida

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: catalog/dags/common/slack.py [hunk 1]
```diff
@@ -396,15 +396,15 @@ def on_failure_callback(context: dict) -> None:
...
+*Exception*: {exception}
 """
 
     message = f"""
 *DAG*: `{dag_id}`
 *Task*: `{task_id}`
 *Logical Date*: {logical_date.strftime('%Y-%m-%dT%H:%M:%SZ')}
-*Log*: {ti.log_url}
+*Log*: <{ti.log_url}|View Logs>
 {exception_message}
```
Comment (dhruvkb) [inline catalog/dags/common/slack.py:408]: This is not directly related to this PR, but why is `exception_message` interpolated here and not written out in this f-string directly? I don't see it being used anywhere else.

**Review by dhruvkb — APPROVED** [review]
LGTM! This is a very welcome change, thanks for the contribution @nicoepp! Also could you leave a comment on the linked issue so that I can assign it to you? (GitHub doesn't allow setting an assignee that has not participated in the issue comments.)

**Comment by dhruvkb** [comment]
This is what the updated error messages look like in Slack.
image

**Comment by zackkrida** [comment]
If you don't mind @stacimc or @AetherUnbound I'd appreciate a cursory review from either of you on this one. No rush!

**Review by stacimc — APPROVED** [review]
Looks great to me, thanks @nicoepp, and +1 for the idea to streamline these Slack messages. Inspired me to add a similar change to [one of my open PRs](https://github.com/WordPress/openverse/pull/4074/files) 😃

**Comment by nicoepp** [comment]
Yay! Thank you @stacimc and @zackkrida!
